src/package-linker.js

```diff
@@ -446,6 +446,24 @@ export default class PackageLinker {
       }
       const ref = pkg._reference;
       invariant(ref, 'Package reference is missing');
+      // TODO: We are taking the "shortest" ref tree but there may be multiple ref trees with the same length
+      const refTree = ref.requests.map(req => req.parentNames).sort((arr1, arr2) => arr1.length - arr2.length)[0];
+      const refTreeLen = refTree.length;
```

Comment:
I'd just use `refTree.length`, it looks like a premature optim (same for `pkgParentsLen`)

Reply:
It was mostly for ease of reading for myself but sure.

Reply:
Oh well, these were easier on the eyes in a previous version of the code. Now they are useless as you pointed out.